src/core/lib/gprpp/map.h

```diff
@@ -117,7 +118,8 @@ class Map {
   iterator end() { return iterator(this, nullptr); }
 
   iterator lower_bound(const Key& k) {
-    key_compare compare;
+    key_compare compare_tmp;
```

Comment:
Why not just declare this as `const` to begin with?

Reply:
This is because one of compiler doesn't build it when it just uses `const key_compare`. [ref](https://stackoverflow.com/questions/7411515/why-does-c-require-a-user-provided-default-constructor-to-default-construct-a)

Reply:
Weird.

Please add a comment explaining this, since it's not at all obvious to the reader.

Reply:
Yeah it's weird. I added the comment.